utils/mocks/create-blocklock-request.ts

```diff
@@ -0,0 +1,278 @@
+            c1: BigInt("0x2863e20a5125b098108a5061b31f405e16a069e9ebff60022f57f4c4fd0237bf"),
+        },
+        y: {
+            c0: BigInt("0x193513dbe180d700b189c529754f650b7b7882122c8a1e242a938d23ea9f765c"),
+            c1: BigInt("0x11c939ea560caf31f552c9c4879b15865d38ba1dfb0f7a7d2ac46a4f0cae25ba"),
+        },
+    };
+
+    const network = await provider!.getNetwork();
+    const chainId = network.chainId;
+
+    const BLOCKLOCK_IBE_OPTS: IbeOpts = createBlocklockIbeOpts(chainId);
+
+    const ct = encrypt_towards_identity_g1(encodedMessage, identity, blocklock_default_pk, BLOCKLOCK_IBE_OPTS);
+
+    // compute gas price
+
+    const feeData = await provider!.getFeeData();
+
+    // feeData.gasPrice: Legacy flat gas price (used on non-EIP-1559 chains like Filecoin or older EVMs)
+    const gasPrice = feeData.gasPrice!;
+
+    // feeData.maxFeePerGas: Max total gas price we're willing to pay (base + priority), used in EIP-1559
+    const maxFeePerGas = feeData.maxFeePerGas!;
+
+    // feeData.maxPriorityFeePerGas: Tip to incentivize validators (goes directly to them)
+    const maxPriorityFeePerGas = feeData.maxPriorityFeePerGas!;
+
+    const isFilecoin = Number(chainId) === 314 || Number(chainId) === 314159;
+
+    let callbackGasLimit = 5000000n;
+
+    let txGasPrice: bigint;
+    let filecoinllbackGasLimitBuffer = 400n;
+    if (isFilecoin) {
+        // Use legacy gasPrice directly
+        txGasPrice = gasPrice > 0 ? gasPrice * 10n : (maxFeePerGas + maxPriorityFeePerGas) * 10n;
+        callbackGasLimit = callbackGasLimit * filecoinllbackGasLimitBuffer;
+    } else {
+        // Use effective gas price based on EIP-1559
+        txGasPrice = maxFeePerGas + maxPriorityFeePerGas;
+        callbackGasLimit = callbackGasLimit;
+    }
+
+    // make direct funding request with enough callbackGasLimit to cover BLS operations in call to decrypt() function
+    // in receiver contract and msg.value to cover the request price + a buffer
+
+    const requestPrice = await blocklockSenderInstance.estimateRequestPriceNative(
+        callbackGasLimit,
+        txGasPrice
+    );
+
+    const bufferPercent = isFilecoin ? 300n : 10n;
+    const valueToSend = requestPrice + (requestPrice * bufferPercent) / 100n;
+
+    console.log("Native / ETH to pay for request", ethers.formatEther(valueToSend));
+
+    const estimatedGas = await mockBlocklockReceiverInstance.createTimelockRequestWithDirectFunding.estimateGas(callbackGasLimit, encodedCondition, encodeCiphertextToSolidity(ct),
+        isFilecoin
+            ? { value: valueToSend, gasPrice: txGasPrice }
+            : {
+                value: valueToSend, maxFeePerGas,
+                maxPriorityFeePerGas,
+            });
+    console.log("estimated gas", estimatedGas);
+
+    const tx = await mockBlocklockReceiverInstance
+        .connect(signer)
+        .createTimelockRequestWithDirectFunding(callbackGasLimit, encodedCondition, encodeCiphertextToSolidity(ct),
+            isFilecoin
```

Comment:
I think Filecoin also supports EIP-1559 tx.

Reply:
okay, I'll test and update the script